### PR TITLE
New JavaDSL for Akka HTTP finally there (2)

### PR DESCRIPTION
**Pull request by johanandren**
Superseeding #20505 

Some squashing and ammending with better commit messages for cleaner history when we merge.

### CODE REVIEW

**Comment by johanandren** [comment]
Resolves #19632 #19519

**Comment by patriknw** [comment]
looks like a good commit history

**Comment by ktoso** [comment]
Looks nice, let's hope validation will finally be happy. ConsistencySpec AFAIR should be passing, that's pretty much the last thing I fixed yesterday.

**Comment by johanandren** [comment]
Argh. The consistency spec isn't picking up the `@CorrespondsTo` annotation on the jenkins but it does locally.

**Comment by ktoso** [comment]
Very weird... perhaps making a clean on jenkins is needed?

**Comment by johanandren** [comment]
_sigh_, unrelated test failure, `ActorPublisherDocSpec`

**Comment by johanandren** [comment]
PLS BUILD

File: akka-http/src/main/scala/akka/http/javadsl/server/PathMatchers.scala [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+  /**
+   * A PathMatcher that matches a single slash character ('/').
+   */
+  def slash(): PathMatcher0 = Slash
+
+  /**
+   * A PathMatcher that matches a single slash character ('/').
+   */
+  def pathEng(): PathMatcher0 = new PathMatcher0(SPathMatchers.PathEnd)
```
Comment (2beaucoup) [inline akka-http/src/main/scala/akka/http/javadsl/server/PathMatchers.scala]: `pathEnd`

File: akka-http/src/main/scala/akka/http/javadsl/server/PathMatchers.scala [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+  private[this] val HexIntegerSegment: PathMatcher1[java.lang.Integer] = fromScala1(SPathMatchers.HexIntNumber.map { i ⇒ i: java.lang.Integer })
+  private[this] val HexLongSegment: PathMatcher1[java.lang.Long] = fromScala1(SPathMatchers.HexLongNumber.map { i ⇒ i: java.lang.Long })
+  private[this] val DoubleSegment: PathMatcher1[java.lang.Double] = fromScala1(SPathMatchers.DoubleNumber.map { i ⇒ i: java.lang.Double })
+  private[this] val UUIDSegment: PathMatcher1[UUID] = fromScala1(SPathMatchers.JavaUUID)
+
+  private[this] val Neutral = fromScala0(SPathMatchers.Neutral)
+  private[this] val Slash = new PathMatcher0(SPathMatchers.Slash)
+  private[this] val PathEnd = new PathMatcher0(SPathMatchers.PathEnd)
+  private[this] val Remaining = new PathMatcher1[String](SPathMatchers.Remaining)
+  private[this] val RemainingPath = new PathMatcher1[Path](SPathMatchers.RemainingPath)
```
Comment (2beaucoup) [inline akka-http/src/main/scala/akka/http/javadsl/server/PathMatchers.scala]: unused?

**Comment by johanandren** [comment]
Maybe I should have let those two fixes slip now that it was green and fix them later on master. Well well, another 2 hours then.

**Comment by 2beaucoup** [comment]
Sorry @johanandren! I'll review later and PR if necessary.

**Comment by johanandren** [comment]
No worries, thanks for looking at it. It's painfully massive so we will likely find a thing or two after merging it.

**Comment by ktoso** [comment]
OH MY GAWD! Tests green finally!
Mind if I push the button @johanandren?

**Comment by johanandren** [comment]
Go! :shipit:

**Comment by ktoso** [comment]
Seems like a few unrelated errors in there...

**Comment by ktoso** [comment]
PLS BUILD

**Comment by ktoso** [comment]
Finally :-)

**Comment by ktoso** [comment]
Thanks everyone for the help here – was a huge effort :-)

**Comment by ktoso** [comment]
Resolves #19632 #19519

**Comment by ktoso** [comment]
Resolves https://github.com/akka/akka/issues/18919

**Comment by jypma** [comment]
OMG it's happening!

**Comment by patriknw** [comment]
Woot! Great work @ktoso, @johanandren and @jypma

**Pull request merged**